src/modules/submissions.js

```diff
@@ -1135,12 +1135,27 @@
 			} );
 		}
 
+		function checkForBlocks() {
+			var deferred = $.Deferred();
```

Comment:
here too, explicit use of $.Deferred() can be removed, by putting a `return` on the next two lines